File: NapakalakiRuby/lib/treasure.rb
# By: Miguel Robles Urquiza
#     Jesús Sánchez de Lechina Tejada

#encoding: utf-8

require_relative "treasure_kind"


class Treasure
  
  def initialize (n, bonus, t)
     @name = n
     @bonus = bonus
     @type = t
    end
    def getName()
      @name
  
    end
        
    def getBonus
      @bonus
    end
    
    def getType
      @type
    end
    
    def to_s
      "Treasure: #{@name} Type: #{@type} Bonus:{@bonus}\n" 
    end
    
end

File: Napakalaki Java/src/NapakalakiGame/Treasure.java
package NapakalakiGame;

/**
 *
 * @author Jesús Sánchez de Lechina Tejada & Miguel Ángel Robles Urquiza
 */
class Treasure {
    private String name;
    private int bonus;
    private TreasureKind treasurekind;
    
    public Treasure(String name, int bonus, TreasureKind t){
        this.name = name;
        this.bonus = bonus;
        this.treasurekind = t;
    }
    /*
    public Treasure(Treasure t){
        this.name = t.getName();
        this.bonus = t.getBonus();
        this.treasurekind = t.getType();
    }
    */
    public String getName(){ return name; }    
    public int getBonus(){ return bonus; }    
    public TreasureKind getType(){ return treasurekind; }
    
     @Override
    public String toString() {
        return "Treasure: "  + name + " Type: " + bonus + " Bonus: " + treasurekind + '\n';
    }
}
